dbms/src/Storages/PathPool.cpp

```diff
@@ -516,6 +516,26 @@ void StableDiskDelegator::addDTFile(UInt64 file_id, size_t file_size, std::strin
     pool.global_capacity->addUsedSize(path, file_size);
 }
 
+bool StableDiskDelegator::updateDTFileSize(UInt64 file_id, size_t file_size)
+{
+    auto iter = pool.dt_file_path_map.find(file_id);
```

Comment:
these codes access to `pool.dt_file_path_map` should be protected by `std::lock_guard lock{pool.mutex};`